subsys/bluetooth/host/mesh/proxy.c

```diff
@@ -566,6 +566,16 @@ static ssize_t prov_ccc_write(struct bt_conn *conn,
 	return len;
 }
 
+static ssize_t prov_ccc_read(struct bt_conn *conn,
+			     const struct bt_gatt_attr *attr,
+			     void *buf, u16_t len, u16_t offset)
+{
+	u16_t value;
+
+	return bt_gatt_attr_read(conn, attr, buf, len, offset, &value,
+				 sizeof(value));
```

Comment:
bt_gatt_attr_read() will read from the pointer given to it in the second last parameter. Here you're giving it an uninitialized variable to read from, so that can't be right.

Reply:
Usually we would set the value in the attr->user_data that way you can synchronize with what has been written.